pallets/dapp-staking/src/types.rs

```diff
@@ -1137,6 +1205,65 @@ impl SingularStakingInfo {
         result
     }
 
+    /// Transfers stake between contracts while maintaining subperiod-specific allocations, era consistency, and bonus-safe conditions.
+    pub fn move_stake(
+        &mut self,
+        destination: &mut SingularStakingInfo<MaxBonusMoves>,
+        amount: Balance,
+        current_era: EraNumber,
+        subperiod: Subperiod,
+    ) -> (Vec<(EraNumber, Balance)>, StakeAmountMoved) {
+        let staked_snapshot = self.staked;
+        let era_and_amount_pairs = self.unstake(amount, current_era, subperiod);
+
+        let voting_stake_moved = staked_snapshot.voting.saturating_sub(self.staked.voting);
+        let build_earn_stake_moved = staked_snapshot
+            .build_and_earn
+            .saturating_sub(self.staked.build_and_earn);
+
+        // Similar to a stake on destination but for the 2 subperiods
+        destination.previous_staked = destination.staked;
+        destination.previous_staked.era = current_era;
+        if destination.previous_staked.total().is_zero() {
+            destination.previous_staked = Default::default();
+        }
+        destination
+            .staked
+            .add(voting_stake_moved, Subperiod::Voting);
+        destination
+            .staked
+            .add(build_earn_stake_moved, Subperiod::BuildAndEarn);
+
+        // Moved stake is only valid from the next era so we keep it consistent here
+        destination.staked.era = current_era.saturating_add(1);
+
+        (
+            era_and_amount_pairs,
+            StakeAmountMoved {
+                voting: voting_stake_moved,
+                build_and_earn: build_earn_stake_moved,
+            },
+        )
+    }
```

Comment:
IMHO, this method should be removed.

The `move` operation is a combination of `unstake` and `stake`.
Nothing more.

We should reuse that code as best we can.
If it means modifying the methods footprint(s) - it's fine.
We have plenty of tests for stake & unstake operations, and these should continue to work as they have before.